js/piwik.js

```diff
@@ -7706,6 +7712,32 @@ if (typeof window.Piwik !== 'object') {
         // initialize the Piwik singleton
         addEventListener(windowAlias, 'beforeunload', beforeUnloadHandler, false);
 
+        window.addEventListener('message', function(e) {
+            try {
+                var data = JSON.parse(e.data);
+            } catch (e) {
+                return;
+            }
+
+            if (typeof data.loaded != 'undefined') {
+                // We received a response to our message sending the initial state to the iframe
+                // We can clear the interval so that we don't send any more
+                stopInitializationInterval();
+                return;
+            }
+
+            if (typeof data.opted_in == 'undefined') {
```

Comment:
btw we have some method for this `isDefined(data.opt_in)` or so is the name

Reply:
Sweet, using this throughout now